lib/Serialization/SerializeSIL.cpp

```diff
@@ -392,10 +392,7 @@ void SILSerializer::writeSILFunction(const SILFunction &F, bool DeclOnly) {
   // Assign a value ID to each SILInstruction that has value and to each basic
   // block argument.
   unsigned ValueID = 0;
-  llvm::ReversePostOrderTraversal<SILFunction *> RPOT(
-      const_cast<SILFunction *>(&F));
-  for (auto Iter = RPOT.begin(), E = RPOT.end(); Iter != E; ++Iter) {
-    auto &BB = **Iter;
+  for (const SILBasicBlock &BB : F) {
```

Comment:
Again, this is how it was even before my patches, e.g. see here: https://github.com/apple/swift/blob/ae8b2ba0ded42cdf9c2f4e1d0bd38d2a2f3ea8eb/lib/Serialization/SerializeSIL.cpp

But I agree that it is kind of using a different style. So, let's use auto here as well.